Source/objects.cpp

```diff
@@ -4350,6 +4351,53 @@ bool Object::IsDisabled() const
 	return IsAnyOf(static_cast<shrine_type>(_oVar1), shrine_type::ShrineFascinating, shrine_type::ShrineOrnate, shrine_type::ShrineSacred);
 }
 
+Object *ObjectAtPosition(Point position, bool considerLargeObjects)
+{
+	if (!InDungeonBounds(position)) {
+		return nullptr;
+	}
+
+	auto objectId = dObject[position.x][position.y];
+
+	if (objectId > 0 || (considerLargeObjects && objectId != 0)) {
+		return &Objects[abs(objectId) - 1];
+	}
+
+	// nothing at this position, return a nullptr
+	return nullptr;
+}
+
+bool IsObjectAtPosition(Point position)
+{
+	return ObjectAtPosition(position) != nullptr;
+}
+
+bool IsItemBlockingObjectAtPosition(Point position)
+{
+	Object *object = ObjectAtPosition(position);
+	if (object != nullptr && object->_oSolidFlag) {
+		// solid object
+		return true;
+	}
+
+	object = ObjectAtPosition(position + Direction::South);
+	if (object != nullptr && object->_oSelFlag != 0) {
+		// open single tile door
+		return true;
+	}

```

Comment:
Previously it was possible to drop items in open doors, wouldn't this change that?

Reply:
This doesn't change the behaviour, you can still drop an item in an open door. You can't drop an item in the tile the door opens into (whether it's open or closed) which appears to be what this check is looking at, though I'm not sure why the existing code was looking at _oSelFlag...

Reply:
And after testing this I realised I got the context for this completely wrong. These checks are looking for interactive objects (breakable barrels and unopened containers). I'll update the comments.

Reply:
_oSelFlag controls how many tiles the object be selectable in. So yeah it makes sens that it would prevent you from dropping items behind the area where you would instead be only able to height light the object and not the item.